### PR TITLE
Select background

### PR DESCRIPTION
**Pull request by Simon-Laux**
- [x] Logic
- [x] Styling
- [x] Translation Strings

### CODE REVIEW

**Comment by Simon-Laux** [comment]
Ah we still need to decide on the default background on all platforms for next release @r10s @Jikstra
If we put in the new one("potato-stamps") on android too or if we keep the old one and decide at the next release.
It would bring some visual change for users to see.

**Comment by r10s** [comment]
tbh, i would prefer a decision on that not now. as said, however, the potato-stamp is not my favorite, but when everyone likes it, we can go for it. but it could be technically better.

but maybe we can also use different backgrounds in the upcoming release and sync them later. wfiw, ios currently has _no_ background at all.

File: src/main/ipc.js [hunk 1]
```diff
@@ -173,11 +173,34 @@ function init (cwd, state, logHandler) {
...
   })
 
-  ipcMain.on('updateDesktopSetting', (e, key, value) => {
+  const updateDesktopSetting = (e, key, value) => {
     const { saved } = app.state
     saved[key] = value
     app.saveState({ saved })
     render()
+  }
+  ipcMain.on('updateDesktopSetting', updateDesktopSetting)
```
Comment (ralphtheninja) [inline src/main/ipc.js]: Maybe do

```js
 ipcMain.on('updateDesktopSetting', (e, k, v) => updateDesktopSetting(k, v))
```

File: src/main/ipc.js [hunk 2]
```diff
@@ -173,11 +173,34 @@ function init (cwd, state, logHandler) {
     e.returnValue = app.localeData
   })
 
-  ipcMain.on('updateDesktopSetting', (e, key, value) => {
+  const updateDesktopSetting = (e, key, value) => {
```
Comment (ralphtheninja) [inline src/main/ipc.js]: Maybe remove `e` since we don't use it?

File: src/main/ipc.js [hunk 3]
```diff
@@ -173,11 +173,34 @@ function init (cwd, state, logHandler) {
...
+    }, (filenames) => {
+      if (!filenames) { return }
+      log.info('BG-IMG Selected File:', filenames[0])
+      const newPath = path.join(getConfigPath(), 'background' + path.extname(filenames[0]))
+      fs.copyFile(filenames[0], newPath, (err) => {
+        if (err) {
+          log.error('BG-IMG Copy Failed', err)
+          return
+        }
+        updateDesktopSetting(null, 'chatViewBgImg', `url("${newPath}")`)
```
Comment (ralphtheninja) [inline src/main/ipc.js]: .. and if so, remove `null` here :)

**Comment by OzancanKaratas** [comment]
I saw some typos. However, I cannot create pull request because commits are conflicting. Please manually merge [this commit](https://github.com/OzancanKaratas/deltachat-desktop/commit/846b561d1aae9b791ef0d0216d0e7c0e2e92d206.patch).

**Comment by ralphtheninja** [comment]
> I saw some typos. However, I cannot create pull request because commits are conflicting. Please manually merge [this commit](https://github.com/OzancanKaratas/deltachat-desktop/commit/846b561d1aae9b791ef0d0216d0e7c0e2e92d206.patch).

Why are commits conflicting? It should work fine if you rebase your work on latest master or on this branch?

**Comment by Simon-Laux** [comment]
@OzancanKaratas please make a new pr for this, It's not changing many typos that were introduced in this pr from what I can see with my sleepy eyes. Also the translations get also fixed on transifex sooner or later so I don't even know if it's worth the effort to do it right now.

File: src/main/ipc.js (hunk 1 above)
Comment (Simon-Laux) [reply src/main/ipc.js]: a new function... maybe more clearer, but us the definition is above it should be fine

File: scss/_global.scss [hunk 4]
```diff
@@ -136,3 +136,21 @@ code {
   image-rendering: -o-crisp-edges;
   image-rendering: pixelated;
 }
+
+
+// ChatView Background selector
+
+.background-preview {
```
Comment (Jikstra) [inline scss/_global.scss]: Please move into scss/modules/_settings.scss

File: src/renderer/components/ChatView.js [hunk 5]
```diff
@@ -242,36 +243,44 @@ class ChatView extends React.Component {
...
-        <Composer
-          ref={this.refComposer}
-          chatId={chat.id}
-          draft={chat.draft}
-          onSubmit={this.writeMessage}
-          setComposerSize={this.setComposerSize.bind(this)}
-        />
-      </ChatViewWrapper>
+      <SettingsContext.Consumer>
+        {(settings) => (
```
Comment (Jikstra) [inline src/renderer/components/ChatView.js]: Is it possible to use hooks here?

**Pull request merged**